### DISCUSSION TITLE
Add custom format of string and custom validation.

### DISCUSSION THREAD

**Discussion by gespispace** [koxudaxi/datamodel-code-generator#426] [Q&A] [answered]
Hello
How do I add additional formatting to a string?

I need format mapping to my class.
Example.
I am making a spec.

```yaml
openapi: 3.0.0
components:
  schemas:
    Network:
      type: object
      properties:
        portRange:
          type: string
          format: port-range
          example: "55-1000"
```

And I want to get it in the end

```py
class Network(BaseModel):
    portRange: Optional[PortRange] = Field(
        None, example='55-1000'
    )
```
Where `PortRange` is my custom validation class.
How to do it?

How can I add my [own validations](https://pydantic-docs.helpmanual.io/usage/validators/#reuse-validators) to the schema, so as not to lose the model with validations in case of re-creation?

Thanks for the help.

**Comment by koxudaxi** [comment, marked as answer]
@gespispace 
> Where PortRange is my custom validation class.
> How to do it?

Currently, it is not supported.

We are talking about custom type in https://github.com/koxudaxi/datamodel-code-generator/discussions/418

> How can I add my own validations to the schema, so as not to lose the model with validations in case of re-creation?

Also, `validators` is not supported too.
But, you can give the code generator a custom template in jinja2. 
https://github.com/koxudaxi/datamodel-code-generator/blob/master/datamodel_code_generator/model/template/pydantic/BaseModel.jinja2

My idea is ...
If a special attribute will be supported in Jsonchema then we can check the value in a custom template :thinking: 


```yaml
openapi: 3.0.0
components:
  schemas:
    Network:
      type: object
      properties:
        name:
          type: string
          customValidator: "padding"
```


```jinja2
...
{%- for field in fields -%}
    {%- if field.field %}
    {{ field.name }}: {{ field.type_hint }} = {{ field.field }}
    {%- elif field.required or (field.represented_default == 'None' and field.strip_default_none)  %}
    {{ field.name }}: {{ field.type_hint }}
    {%- else %}
    {{ field.name }}: {{ field.type_hint }} = {{ field.represented_default }}
    {%- endif %}
    {%- if field.customValidator ==  'format_name' %}
    @validator('{{ field.name}}')
    def padding_{{ field.name }}(value):
       return value.ljust(10, ' ')
    {%- endif %}
 ...
```
